src/Utils/FileSystem.php

```diff
@@ -131,10 +131,11 @@ public static function read(string $file): string
 
 	/**
 	 * Writes a string to a file.
+	 * @param int|NULL $mode
```

Comment:
It does not look obvious what this parameter does... Is that creation mode or does it override current file mode?

Maybe add to docs something like:
- Alters file mode to given value (does not matter if file already or will be created exists)

**Shouldn't NULL be the default value?** Current default value is not secure at all. It allows everyone on the server to read and write to this file. And imagine situation where that server is shared... Why not to leave this value to `NULL` and keep responsibility of setting proper permissions on system admin that configured that server? In my case `chmod(0666)` basically breaks what was set by ACL and allows reading and writing by everyone.